### PR TITLE
Adjust the Heroku, Release, and Assets addon to make the app deployable 

### PR DESCRIPTION
**Pull request by andyduong1920**
## What happened

The new generate project can't deploy to Heroku

## Insight

1/ [All Heroku Postgres client connections require SSL](https://devcenter.heroku.com/changelog-items/2035)

I have faced that error when testing the template

Screen Shot 2022-03-03 at 13 41 14

Following the guide on https://hexdocs.pm/phoenix/heroku.html

image

Although we have some documentation for that in the Readme, and it requires some manual step to config, as the template we could make it ready to deploy by automate some step.

2/ Adjust the Release addon as there are some changes in the Elixir 1.13.3 and Phoenix 1.6.6 for the Runtime configuration part.

3/ Remove the `priv/static` in `.dockerignore` because in Phoenix 1.6, all asserts like images, fonts, will be placed in `priv/static` folder.

image

If we don't allow the `priv/static` folder there, the assets like images, fonts won't be copy and it's not on Production.

image

## Proof Of Work

The test is passed, and the application can deployed to Heroku

1/ Test is passed https://github.com/andyduong1920/demo-elixir-template-4-0/actions/runs/1927159808

2/ Heroku app https://demo-elixir-template-4-0.herokuapp.com/

### CODE REVIEW

File: test/nimble_template/addons/github_test.exs [hunk 1]
```diff
@@ -287,6 +287,22 @@ defmodule NimbleTemplate.Addons.GithubTest do
         assert_file(".github/workflows/deploy_heroku.yml")
       end)
     end
+
+    test "enables ssl for Ecto", %{
```
Comment (topnimble) [inline test/nimble_template/addons/github_test.exs:291]:
```suggestion
    test "enables SSL for Ecto", %{
```

**Review by topnimble — APPROVED** [review]

File: lib/nimble_template/addons/variants/phoenix/mix_release.ex [hunk 2]
```diff
@@ -23,4 +25,61 @@ defmodule NimbleTemplate.Addons.Phoenix.MixRelease do
 
     project
   end
+
+  defp edit_files(%{otp_app: otp_app, web_module: web_module} = project) do
+    Generator.delete_content(
+      "config/runtime.exs",
+      """
+      # Start the phoenix server if environment is set and running in a release
```
Comment (longnd) [inline lib/nimble_template/addons/variants/phoenix/mix_release.ex:33]: do you mean like this?
```suggestion
      # Start the phoenix server if the `PHP_SERVER` environment variable is set and running in a release
```

File: lib/nimble_template/addons/variants/phoenix/mix_release.ex [hunk 3]
```diff
@@ -23,4 +25,61 @@ defmodule NimbleTemplate.Addons.Phoenix.MixRelease do
...
+    )
+
+    Generator.delete_content(
+      "config/runtime.exs",
+      """
+
+        # ## Using releases
+        #
+        # If you are doing OTP releases, you need to instruct Phoenix
+        # to start each relevant endpoint:
```
Comment (longnd) [inline lib/nimble_template/addons/variants/phoenix/mix_release.ex:48]: I think we can rephase this sentence to remove `you` as it isn't appropriate in technical comments 
```suggestion
        # For OTP releases, instruct Phoenix to start each relevant endpoint:
```

File: lib/nimble_template/addons/variants/phoenix/mix_release.ex [hunk 4]
```diff
@@ -23,4 +25,61 @@ defmodule NimbleTemplate.Addons.Phoenix.MixRelease do
...
+        #
+        #     config :#{otp_app}, #{web_module}.Endpoint, server: true
+        #
+        # Then you can assemble a release by calling `mix release`.
```
Comment (longnd) [inline lib/nimble_template/addons/variants/phoenix/mix_release.ex:52]:
```suggestion
        # Then, assemble a release by calling `mix release`.
```

**Review by longnd — APPROVED** [review]
Just some minor comments

File: lib/nimble_template/addons/variants/phoenix/mix_release.ex (hunk 2 above)
Comment (andyduong1920) [reply lib/nimble_template/addons/variants/phoenix/mix_release.ex:33]: This is the delete content command @longnd and that content comes from the Phoenix :D

File: lib/nimble_template/addons/variants/phoenix/mix_release.ex (hunk 3 above)
Comment (andyduong1920) [reply lib/nimble_template/addons/variants/phoenix/mix_release.ex:48]: same above, it comes from the default Phoenix @longnd :D we just remove it

File: lib/nimble_template/addons/variants/phoenix/mix_release.ex (hunk 4 above)
Comment (andyduong1920) [reply lib/nimble_template/addons/variants/phoenix/mix_release.ex:52]: same as above :D

**Pull request merged**